### DISCUSSION TITLE
Upgraded to 14.2.2 - ReferenceError: _nextOriginalFetch is not defined

### DISCUSSION THREAD

**Discussion by damyco** [vercel/next.js#64713] [App Router] [answered]
Hi, I have upgraded my Next.js app to the most recent version and now getting this error when trying to build the app and deploy on Vercel:
`ReferenceError: _nextOriginalFetch is not defined`
I'm using _nextOriginalFetch to fetch products from my backend and be able to pass header with lang as normal fetch would not pass the headers at all in server side component for some reason... 
```
const productData = await _nextOriginalFetch(`${apiUrl}/products`, {headers: {'Accept-Language': lang}});
```
Did something change? Can I now pass header using fetch? How can I fix this?

This is preventing me from using the latest version and there is a feature added that I really need.

**Comment by icyJoseph** [comment]
Hi,

I guess `_nextOriginalFetch` was moved, or renamed... or just not injected into the global scope, anymore.

See this extract, from https://github.com/vercel/next.js/discussions/56446:

> _nextOriginalFetch is a Next-specific internal flag. No third-party consumers should depend on such implementation details, it's unsafe.

Can you try to verify again that headers are not passed when using the exposed fetch function?

**Comment by damyco** [comment]
> Hi,
> 
> I guess `_nextOriginalFetch` was moved, or renamed... or just not injected into the global scope, anymore.
> 
> See this extract, from #56446:
> 
> > _nextOriginalFetch is a Next-specific internal flag. No third-party consumers should depend on such implementation details, it's unsafe.
> 
> Can you try to verify again that headers are not passed when using the exposed fetch function?

I have swapped for fetch and it doesn't work in local dev, it also doesn't work if I run `build` and then `start`.
It does work in production (Vercel preview) though... 

How can I make it work in local dev or build? If I have similar feature to build reliant on fetching products I won't be able to test it as it always fetches it in English for some reason, completely ignoring the header.

**Comment by damyco** [comment, marked as answer]
I found what was the issue, my local backend did not had translation files available - I upgraded my machine to a new one and forgot to move these. 🤦🏻 Apologies!

I can confirm the standard `fetch()` now can use headers on server side and works perfectly locally and in production.

Thanks, I'm glad I don't have to use `_nextOriginalFetch` anymore 👍
